plumpy/process_states.py

```diff
@@ -229,7 +230,7 @@ async def execute(self) -> State:  # type: ignore # pylint: disable=invalid-over
                     result = self.run_fn(*self.args, **self.kwargs)
                 finally:
                     self._running = False
-            except Interruption:
+            except (Interruption, asyncio.CancelledError):
```

Comment:
maybe it makes sense to also here make a separate case for `asyncio.CancelledError` and add the multi-line comment + add a note that this can be removed once support for python 3.7 is dropped